shadowsocks/tcp.go

```diff
@@ -259,7 +269,11 @@ func (s *tcpService) handleConnection(listenerPort int, clientConn onet.DuplexCo
 		logger.Debugf("TCP Error: %v: %v", connError.Message, connError.Cause)
 		status = connError.Status
 	}
-	s.m.AddClosedTCPConnection(clientLocation, keyID, status, proxyMetrics, timeToCipher, connDuration)
+	var id string
+	if cipherEntry != nil {
```

Comment:
Yes, this is equivalent to `if keyErr != nil`.  This code runs whether `keyErr` is nil or not.  If it would be clearer, we could move this to line 234.